core/src/main/java/hudson/model/Run.java

```diff
@@ -660,32 +671,43 @@ public final long getStartTimeInMillis() {
     }
 
     @Exported
+    @CheckForNull
     public String getDescription() {
         return description;
     }
 
 
     /**
      * Returns the length-limited description.
+     * The method tries to take the HTML tags into account, but it is a best-effort attempt.
+     * Also, the method will not work properly if a non-HTML {@link hudson.markup.MarkupFormatter} is used.
      * @return The length-limited description.
-     * @deprecated truncated description uses arbitrary and unconfigurable limit of 100 symbols
+     * @deprecated truncated description based on the {@link #TRUNCATED_DESCRIPTION_LIMIT} setting.
      */
     @Deprecated
-    public @Nonnull String getTruncatedDescription() {
-        final int maxDescrLength = 100;
-        if (description == null || description.length() < maxDescrLength) {
+    public @CheckForNull String getTruncatedDescription() {
+        if (TRUNCATED_DESCRIPTION_LIMIT < 0) { // disabled
             return description;
         }
+        if (TRUNCATED_DESCRIPTION_LIMIT == 0) { // Someone wants to suppress descriptions, why not?
```

Comment:
personally I would make 0 disable truncation as 0 often means unlimited, why introduce a feature to disable descriptions? has anyone asked for it?

Reply:
"Possible to hide" in https://issues.jenkins-ci.org/browse/JENKINS-60299

Reply:
That was 2 possible solutions, didn't suggest implementing both 😉

Reply:
Hi, just wanted to chime in. I'm the one who created [JENKINS-60299](https://issues.jenkins-ci.org/browse/JENKINS-60299). For our use-case a possibility to disable the build description in the build history widget would be really nice. Note though that we very much need the build description for each build, just not in the widget itself. 
Off course, I'm just one user so I understand if our specific need cannot be taken into account and implemented as a feature for all users :-)

Reply:
🤷‍♂ fine with me then.

Reply:
Seems like a reasonable option for people using non-HTML markup formatting, which is horrifically broken by the server-side, pre-transformation truncation.